bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/handler/AbstractRingHandler.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ring.handler;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ring.internal.RingDeviceRegistry;
+import org.openhab.binding.ring.internal.errors.DeviceNotFoundException;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractRingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Wim Vissers - Initial contribution
+ * @author Ben Rosenblum - Updated for OH4 / New Maintainer
+ */
+
+@NonNullByDefault
+public abstract class AbstractRingHandler extends BaseThingHandler {
+
+    // Current status
+    protected OnOffType status = OnOffType.OFF;
+    protected OnOffType enabled = OnOffType.ON;
+    protected final Logger logger = LoggerFactory.getLogger(AbstractRingHandler.class);
+
+    // Scheduler
+    protected @Nullable ScheduledFuture<?> refreshJob;
+
+    /**
+     * There is no default constructor. We have to define a
+     * constructor with Thing object as parameter.
+     *
+     * @param thing
+     */
+    public AbstractRingHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing AbstractRingHandler");
+        Map<String, String> properties = editProperties();
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, getThing().getUID().getId());
+        updateProperties(properties);
```

Comment:
That looks weird. Why do you put the thing id as serial number in the properties?